dask/dataframe/shuffle.py

```diff
@@ -128,10 +138,11 @@ def __dask_distributed_pack__(self, client):
             "name_input": self.name_input,
             "meta_input": to_serialize(self.meta_input),
             "parts_out": list(self.parts_out),
+            "annotations": self.expand_annotations(string_keys=True),
```

Comment:
Hrm, I wonder if there is some way to generalize this so that we don't have to handle this for every future layer type.  Thoughts?

Reply:
I thought about this but calling ``self.expand_annotations`` explicitly in ``__dask_distributed_pack__`` is the best option I could think of, *given that the code base is in a state of churn*. In particular, there's a need to call this in ``distributed.protocol.highlevelgraph._materialized_layer_pack`` too, which lies outside the HLG Layer hierarchy.

Based on the [docstrings](https://github.com/dask/dask/blob/master/dask/highlevelgraph.py#L143-L170) I considered changing Layer.__dask_distributed_pack__ to the following:

```python
class Layer:
    def __dask_distributed_pack__(self, client):
       if self.annotations is None:
           return None
     
        return {"annotations": self.expand_annotations()}
```

Then this could work:

```python
class Blockwise(Layer):
    def __dask_distributed_pack__(self, client):
        state = super().__dask_distributed_pack__(client)
        state["other"] = ...
        return state

class SimpleShuffleLayer(Layer):
    def __dask_distributed_pack__(self, client):
        state = super().__dask_distributed_pack__(client)
        state["other"] = ...
        return state
```

Any objections to the above @madsbk and @rjzamora